Recruit-Training/Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -37,13 +37,14 @@ QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 
         if(isCharging())
         {
-            totalHoursRemaining = totalAmpHoursUsed() / current_ * -1;
+            totalHoursRemaining = abs(totalAmpHoursUsed() / current_);
         }
         else
         {
             totalHoursRemaining = (BATTERY_AMP_HOUR_CAPACITY - totalAmpHoursUsed()) / current_;
         }
 
+        //If the totalHoursReamining is greater than 24 continue to subtract it otherwise isn't valid to display since its more than a day
```

Comment:
Misspelled "Remaining" :)

But now I understand that you want to work around the 24 hour limitation of QTime. TBH, I don't think it was intended that this would happen, but I'm not sure about the way it's being addressed here.

Conceptually, there's 2 ways of addressing this issue for hours > 24 that people have been doing, ex 60 hours:
1. Return the number of hours total, so you would display 60:00:00.000
2. Allow it to loop around - you display 12:00:00.000

If you want to do option 1, you would have to store the excess hours in a new variable that is accessed later. All the work in calculating the seconds, hours, and minutes done here would be needed.

Option 2 is fine too, and your implementation works, but it's there's a better way. Look into QTime's `addMSecs()` function, which wraps around as the time passes midnight.

Reply:
Also, a small critique of your comment - `If the totalHoursReamining is greater than 24 continue to subtract it` can be already derived from reading the first line of the while loop. Unless the code isn't straightforward enough to just read, most comments should be supplementary - explaining the intention behind the change, instead of what the change does.

ie.
```
// QTime does not support hours > 24
while(...) 
```